### ISSUE TITLE
Question about the '_gaussian_dist_pdf'

### ISSUE THREAD

**Issue by laycoding** [motokimura/PyTorch_Gaussian_YOLOv3#13]
Hi @motokimura,  
Thanks for your great implementation! :+1:
I'm working on the false positive in SSD, so I transplanted the gaussian part(nll loss) in your repo to replace the smooth-l1 loss. But I found a slight difference with the original paper, you use the net to predict the *sigma_xywh* (<img src="https://render.githubusercontent.com/render/math?math=(\sum t_{x})^2">) , which is represented by var in https://github.com/motokimura/PyTorch_Gaussian_YOLOv3/blob/b51eb917ede2a1d7bc6eb0e220bf330f43baf842/models/yolo_layer.py#L236
Then treat the *(1 - sigma)*, which actually is <img src="https://render.githubusercontent.com/render/math?math=1-(\sum t_{x})^2"> ,like the uncertainty in https://github.com/motokimura/PyTorch_Gaussian_YOLOv3/blob/b51eb917ede2a1d7bc6eb0e220bf330f43baf842/models/yolo_layer.py#L125
After I found that, I use the net to predict the <img src="https://render.githubusercontent.com/render/math?math=\sum t_{x}"> instead of the <img src="https://render.githubusercontent.com/render/math?math=(\sum t_{x})^2">, I got a little better than the original one. 
BTW, have you try the 'sigma_const' in you experiments you mentioned in https://github.com/jwchoi384/Gaussian_YOLOv3/issues/40#issuecomment-559404844, or a better way to use the uncertainty?

**Comment by motokimura** [comment]
Hi @laycoding. 
Sorry for my late response.

Thanks for reporting!
I will to try to reproduce your result and then fix the repo 
(Actually I was not sure this score scaling `pred[..., 4] *= (1.0 - sigma)` is correct because the original paper does not say about it explicitly).

> BTW, have you try the 'sigma_const' in you experiments you mentioned in jwchoi384/Gaussian_YOLOv3#40 (comment), or a better way to use the uncertainty?

Not yet. Your PR (with experiment results) is more than welcome!

**Comment by laycoding** [comment]
@motokimura, Thanks for your response!
I only did the experiments based on our own datasets and different framework(SSD), limited by time, so I don't put the results here.
I think the point is how we treat the sigma and the uc(uncertainty),  we use the net to predict the in \mu Gaussian distribution(not the \mu^2), and set `uc=avg(sigma(\mu))` .
After we tested on real data, we set the `scores *=(1-uc^2)` considering the trade-off between TPs and FPs.
Hope this helps~
